### DISCUSSION TITLE
How to remap a complex behaviour?

### DISCUSSION THREAD

**Discussion by rwijtvliet** [LazyVim/LazyVim#2905] [General]
# Intro

In lazyvim, the `f` key ("find" in default (n)vim, finds first instance of a character to the right of the cursor) and other related movements are remapped to quite complex behaviours. 

This behaviour is defined in `~/.local/share/nvim/lazy/flash.nvim/lua/flash/plugins/char.lua`, which has the following code (and more):

```lua
  -- (...)
  for _, key in ipairs({ "f", "F", "t", "T", ";", "," }) do
    if keys[key] then
      vim.keymap.set({ "n", "x", "o" }, keys[key], function()
        M.jumping = true
        local autohide = Config.get("char").autohide
        if Repeat.is_repeat then
          M.jump_labels = false -- never show jump labels when repeating
          M.state:jump({ count = vim.v.count1 })
          M.state:show()
        else
          M.jump(key)
        end
        vim.schedule(function()
          M.jumping = false
          if M.state and autohide then
            M.state:hide()
          end
        end)
      end, {
        silent = true,
      })
    end
  end
  -- (...)
```

# Task

I want to remap this functionality to another key. 

**Is there a way to do that, without copying this code into my own `mappings.lua` file?**

**Comment by rwijtvliet** [comment]
Ah. I can probably use a custom definition for `folke/flash.nvim` in the `plugins` directory... right? I'll give that a try.

**Reply by dpetka2001** [reply]
Read the options that you can pass to the plugin [here](https://github.com/folke/flash.nvim#%EF%B8%8F-configuration). This will help you achieve what you want. Particularly `char.keys`.
